core/textile/bucket/crypto/crypto.go

```diff
@@ -0,0 +1,177 @@
+package crypto
+
+import (
+	"bytes"
+	b64 "encoding/base64"
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/FleekHQ/space-daemon/log"
+)
+
+func parseKeys(key []byte) (aesKey, iv, hmacKey []byte, err error) {
+	if len(key) != aesKeySize+ivKeySize+hmacKeySize {
+		return nil, nil, nil, errors.New("unsupported encryption keys provided.")
+	}
+
+	return key[:aesKeySize], key[aesKeySize:(aesKeySize + ivKeySize)], key[(aesKeySize + ivKeySize):], nil
+}
+
+// EncryptPathItems returns an encrypted path and a Reader that reads the encrypted data from the
+// plain reader passed into the function.
+// Encrypted data is AES-CTR of data + AES-512 HMAC of encrypted data
+//
+// NOTE: key must be a 64 byte long key
+// To decrypt the result of this function use the DecryptPathItems function
+func EncryptPathItems(key []byte, path string, plainReader io.Reader) (string, io.Reader, error) {
+	// split key into key and secret
+	// use key and secret and IV
+
+	// encrypt path
+	log.Debug("Encrypting Path Items", "path:"+path)
+	aesKey, iv, hmacKey, err := parseKeys(key)
+	if err != nil {
+		return "", nil, err
+	}
+
+	encryptedPath := ""
+	pathParts := strings.Split(path, "/")
+	pathPartsLen := len(pathParts)
+	for i, pathItem := range pathParts {
+		if pathItem == "" {
+			continue
+		}
+		encryptedPathReader, err := NewEncryptReader(
+			strings.NewReader(pathItem),
+			aesKey,
+			iv,
+			hmacKey,
+		)
+		if err != nil {
+			return "", nil, err
+		}
+
+		encryptedPathItem, err := readAsBase64Strings(encryptedPathReader)
+		if err != nil {
+			return "", nil, err
+		}
+		encryptedPath += encryptedPathItem
+		if i != pathPartsLen-1 {
+			encryptedPath += "/"
+		}
+	}
+
+	log.Debug("Encrypting Path Items", "path:"+path, "encryptedPath:"+encryptedPath)
```

Comment:
Maybe we should remove this log when merging because it defeats the purpose of hiding data through encryption.

Reply:
True, I'll remove it